cornice/validators/__init__.py

```diff
@@ -33,7 +34,8 @@ def extract_cstruct(request):
     if request.content_type == 'application/x-www-form-urlencoded':
         body = request.POST.mixed()
 
-    elif request.content_type and request.content_type != 'application/json':
+    elif (request.content_type and
```

Comment:
I would like to suggest an intermediary variable here (like `is_json`) to avoid multiple line `if`.